### PR TITLE
Solve nominal temperature differences in heat exchanger convergence issue

### LINKED ISSUE #458: [FEATURE] Add option to calculate the nominal temperature rise/fall
### What is your precise need ?
To set the `nominal_cold_side_temperature_rise ` and `nominal_hot_side_temperature_drop` for the flue gas HEX can be a challenge. Sometimes we dont have HMBD information about this. So it would be good to add an option to direct calculate it e.g. like this:
`HEX.nominal_cold_side_temperature_rise = HEX.T_hot_in - HEX.T_cold_in;`
`HEX.nominal_hot_side_temperature_drop = HEX.T_hot_in - HEX.T_cold_in;`

### What solution do you see ?
Add a parameter to the HEX components to decide if we want to set the nominal deltaT or calculate it.

### PR DESCRIPTION
**Pull request by nabily-met**
## Goal

Resolves #458 

The goal is to avoid convergence problem related to the values given to the nominal temperature rise and drop in heat exchangers.
A new boolean parameter `nominal_DT_default` is added:
- If `true`, the nominal temperature rise and drop will be equal to the maximum possible temperature difference, `hot_side.T_in - cold_side.T_in`.
- If `false`, customized values should be given.

By default, `nominal_DT_default = true`

⚠️ Breaking change:
You can solve compatibility issue by simply setting `nominal_DT_default` to `false` in all the heat exchangers in your model.
It should be noted, that these temperatures impacts the values of the parameters. If you want to use the default settings after a calibration with customized values, a new calibration should be carried out

## Type of change 

- [ ] Bugfix
- [ ] New feature
- [x] Refactoring change
- [ ] Release & Version Update (don't forget to change the version number in `package.mo`)

Will it break anything in previous models ?

- [x] Breaking change (If yes, make sure to point it out in the changelog)
- [ ] Non-Breaking change

## Checklist

- [x] I have added the appropriate tags, reviewers, projects (and detailed the size and priority of my PR) and linked issues to this PR
- [x] I have performed a self-review of my own code
- [x] I have checked that all existing tests pass
- [ ] I have checked that my work is compatible with [OpenModelica](https://openmodelica.org/)
- [x] I have added/updated tests that prove my development works and does not break anything.
- [ ] I have made corresponding changes or additions to the documentation (in [Notion documentation](https://www.notion.so/metroscope/Metroscope-Modeling-Library-Documentation-MML3-WIP-50c8703c294446059d3b4a70d6ae4a71))
- [x] I have added corresponding entries to the [Changelog](../CHANGELOG.md)
- [x] I have checked for conflicts with target branch, and merged/rebased in consequence

You can also fill these out after creating the PR, but make sure to **check them all before submitting your PR for review**.

### CODE REVIEW

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/FuelHeater.mo [hunk 1]
```diff
@@ -126,6 +129,12 @@ equation
   assert(pinch > 0, "A negative pinch is reached", AssertionLevel.warning); // Ensure a positive pinch
   assert(pinch > 1 or pinch < 0,  "A very low pinch (<1) is reached", AssertionLevel.warning); // Ensure a sufficient pinch
 
+  // Nominal temperature differences
+  if nominal_DT_default then
+    nominal_cold_side_temperature_rise = hot_side.T_in - cold_side.T_in; // Corresponds to the maximum achievable temperature difference
+    nominal_hot_side_temperature_drop = hot_side.T_in - cold_side.T_in; // Corresponds to the maximum achievable temperature difference
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/FuelHeater.mo:135]: There is a copy paste issue, both in equation and in comment here 😇

File: MetroscopeModelingLibrary/Partial/HeatExchangers/WaterFlueGasesMonophasicHX.mo [hunk 2]
```diff
@@ -135,6 +138,12 @@ equation
   // The estimation of the Cp outlet is calculated for an outlet temperature based on the nominal temperature rise of the H&MB diagram.
   // For more details about this hypothesis, please refer the Economiser page of the MML documentation.
 
+  // Nominal temperature differences
+  if nominal_DT_default then
+    nominal_cold_side_temperature_rise = hot_side.T_in - cold_side.T_in; // Corresponds to the maximum achievable temperature difference
+    nominal_hot_side_temperature_drop = hot_side.T_in - cold_side.T_in; // Corresponds to the maximum achievable temperature difference
+  end if;
+
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/Partial/HeatExchangers/WaterFlueGasesMonophasicHX.mo:146]: There is a copy paste issue, both in equation and in comment here 😇

**Review by pierre-eliep-met — CHANGES_REQUESTED** [review]
Just a mistake but otherwise I agree it's a good feature :)

**Comment by pierre-eliep-met** [comment]
Actually, I saw your comment [here](https://github.com/Metroscope-dev/metroscope-modeling-library/issues/458#issuecomment-2464895172), so maybe it was intended actually ?

**Comment by pierre-eliep-met** [comment]
If yes, could you update the comment to explain these unexpected duplicated lines ?

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/FuelHeater.mo (hunk 1 above)
Comment (nabily-met) [reply MetroscopeModelingLibrary/MultiFluid/HeatExchangers/FuelHeater.mo:135]: It is supposed to be the same 🤔

File: MetroscopeModelingLibrary/Partial/HeatExchangers/WaterFlueGasesMonophasicHX.mo (hunk 2 above)
Comment (nabily-met) [reply MetroscopeModelingLibrary/Partial/HeatExchangers/WaterFlueGasesMonophasicHX.mo:146]: The maximum DT applies for both the rise and drop. In other words, the maximum temperature that the cold_side can perform is reaching hot_side.T_in, and the the minimum temperature that the hot_side could reach is cold_side.T_in

**Review by pierre-eliep-met — APPROVED** [review]
Ok!
I tend to think that the best would rather be to have a variable like 
```modelica
maximum_achiveable_temperature_difference = hot_side.T_in - cold_side.T_in;

if nominal_DT_default then
  nominal_cold_side_temperature_rise = maximum_achiveable_temperature_difference;
  nominal_hot_side_temperature_drop = maximum_achiveable_temperature_difference;
```

**Comment by nabily-met** [comment]
> ```modelica
> maximum_achiveable_temperature_difference
> ```

**Pull request merged**